src/InteractiveWindow/Editor/InteractiveWindow_UIThread.cs

```diff
@@ -196,20 +196,21 @@ public UIThreadOnly(InteractiveWindow window, IInteractiveWindowEditorFactorySer
                 _outputTrackingCaretPosition = -1;
             }
 
-            public Task<ExecutionResult> ResetAsync(bool initialize)
+      
+            public Task<ExecutionResult> ResetAsync(bool initialize, bool isFromPrompt)
             {
                 Debug.Assert(State != State.Resetting, "The button should have been disabled.");
 
                 if (_window._stdInputStart != null)
                 {
-                     CancelStandardInput();
+                    CancelStandardInput();
                 }
 
                 _window._buffer.Flush();
 
                 // replace the task being interrupted by a "reset" task:
                 State = State.Resetting;
-                _currentTask = _window._evaluator.ResetAsync(initialize);
+                _currentTask = ((IInteractiveEvaluator2)(_window._evaluator)).ResetAsync(initialize, isFromPrompt);
```

Comment:
I considered that but it needed a change to the constructor and was hence causing changes to several APIs. Since the new interface is only for reset, it is probably fine to leave it as is.